### DISCUSSION TITLE
Removing initramfs

### DISCUSSION THREAD

**Discussion by EasyIP2023** [OE4T/meta-tegra#652] [Q&A] [answered]
Per my understand one can remove initramfs via ```INITRAMFS_IMAGE_${MACHINE}``` variable as explained here https://github.com/OE4T/meta-tegra/issues/383#issuecomment-654975082. Now I have not tried this on branch dunfell-l4t-r32.5.0 but on branch dunfell-l4t-r32.4.3 this resulted in unsuable usb ports.

Is there any other way to load usb control firmware? Asking largely becuase I want to lower boot times. Loading initramfs first then switching to rootfs via switch_root(8) in init_boot.sh slows down the time require for the device to boot.

**Comment by kekiefer** [comment, marked as answer]
Is it enough to install the `tegra-firmware-xusb` package in your normal image?

**Reply by EasyIP2023** [reply]
I'll try that out

**Reply by EasyIP2023** [reply]
Whenever I add ```tegra-firmware-xusb``` and disable initramfs it results in a kernel panic. Even without a custom partition configuration file. When it goes to mount it can't find rootfs for both the xavier agx/nx devkit. I'm going to update the ```KERNEL_ROOTSPEC``` variable to contain  hardcoded absolute path to block dev (```root=/dev/mmcblk0p1```), but that may not fix my issue.
![20210407_173928](https://user-images.githubusercontent.com/8561172/114084778-2c327800-9876-11eb-8acb-1dd09678f8cd.jpg)
![20210408_140713](https://user-images.githubusercontent.com/8561172/114084801-32c0ef80-9876-11eb-91d2-2af614741788.jpg)

**Reply by madisongh** [reply]
Append the `root=...` setting to `KERNEL_ARGS`.  `KERNEL_ROOTSPEC` is only used with u-boot extlinux.conf files.

**Reply by EasyIP2023** [reply]
THANKS!!
